internal/update/update.go

```diff
@@ -31,6 +33,49 @@ type StateEntry struct {
 	LatestRelease      ReleaseInfo `yaml:"latest_release"`
 }
 
+func ShouldCheckForExtensionUpdate() bool {
+	if os.Getenv("GH_NO_UPDATE_NOTIFIER") != "" {
+		return false
+	}
+	if os.Getenv("CODESPACES") != "" {
+		return false
+	}
+	return !IsCI() && IsTerminal(os.Stdout) && IsTerminal(os.Stderr)
+}
+
+func CheckForExtensionUpdate(em extensions.ExtensionManager, ext extensions.Extension, stateFilePath string) (*ReleaseInfo, error) {
+	stateEntry, _ := getStateEntry(stateFilePath)
+	if stateEntry != nil && time.Since(stateEntry.CheckedForUpdateAt).Hours() < 24 {
+		return nil, nil
+	}
```

Comment:
Do you think this interval could get annoying for pinned extension users?

_Disclaimer: I say this as someone who does not leverage the extension pinning feature, I could be misunderstanding something_. 

My guess is that someone pins an extension because they do not want whatever the latest version is. I recognize that this user may still want to know when a new version becomes available, but do we want to continue notifying them at the same 24h interval as other extensions? Would a longer interval make more sense for these users that have already expressed that they do not want to use the latest version?

If we leave pinned extensions as checking and notifying once every 24 hours, I foresee a situation where a user is notified of an available update at most once a day for an extension that they have no intention of updating. In that case, we are giving the user almost no choice but to disable the extension update check altogether if they feel disrupted by the update notification.

Direct Qs:
- Am I correct in thinking that this might be a problem for some folks, and do I understand the pinning feature correctly?
- What do you think about adjusting the interval for pinned extensions?
- Is there something else you could think of to allow a pinned extension user to avoid receiving such frequent notifications, while leaving the updating notification feature on?

Reply:
Perhaps this is a bit spicy, but what if we disable the update checking logic for pinned extensions altogether? 

I can loosely see the logic behind a user expressing that they want `x` version means we shouldn't tell them about available updates. And, as long as that behavior is documented, I can see this making sense to do.

Reply:
Let me think on this question a bit, but here's an alternative view: **users pin a version to ensure a specific SHA is installed.**

Like GitHub Actions, the version is:

- tag / release
- branch
- SHA

Pining an extension is more than casually preventing someone from upgrading the extension &mdash; which can be done in a few ways with `gh ext` commands &mdash; so I'm not 💯 this is meant to avoid upgrading. 🤔

That said, let me re-read the code here versus the `gh` code.  I think in the `gh` code it is actually checking that the version looked up differs from the one that's recorded in the state file, which means users are only notified of new versions.  If that's the case, then we should make sure the extension release checking logic is going the same, so users are only notified when a new version is seen for the first time.